README.md

```diff
@@ -48,33 +65,470 @@ cache read/white process, etc. Command handler can be sync and publish the
 result immediately. On the other hand it's the best place in the
 application to write asynchronous code.
 
-The restriction is to create only one command handler for each command.
+The restriction is to create only one command handler per command.
 
 ### Event
 
-**Command Handler** is responsible for publishing **events** that will be
+Command Handler is responsible for publishing **events** that will be
 consumed by reactions. There are 3 types of events:
 
-- **Progress** which indicates that the operation triggered by command has been
+- `Progress` event indicates that the operation triggered by command has been
 started and is in the pending state at the moment.
-- **Data** which holds the output produced by the command execution
-- **Error** notifies that an error has been occurred during the command
-execution
+- `Data` event holds the output produced by the command execution.
+- `Error` notifies that an error has been occurred during the command
+execution.
 
 ### Reaction
 
 **Reaction** responds to event published by command handler. It is supposed
 to handle 3 possible event types by describing the desired behavior in the
 each scenario:
 
-- **Wait** function reacts on `Progress` type of the event
-- **Consume** function reacts on `Data` type of the event.
-- **Rescue** function is a fallback for the case when `Error` type of the event
-has been received.
+- `Wait` function reacts on `Progress` type of the event
+- `Consume` function reacts on `Data` type of the event.
+- `Rescue` function is a fallback for the case when `Error` event has been
+received.
 
 Normally reaction performs UI updates, but could also be used for other kinds
 of output processing.
 
+## The flow
+
+### Command execution
+
+The first step is to declare a command. Your command type has to conform to
+`Aftermath.Command` protocol and the `Output` type must be implicitly specified.
+
+Let's say we want to fetch a list of books from some untrusted resource and
+correct typos in titles and author names 🤓.
+
+```swift
+// This is our model we are going to work with.
+struct Book {
+  let id: Int
+  let title: String
+  let author: String
+}
+
+struct BookListCommand: Command {
+  // Result of this command will be a list of books.
+  typealias Output = [Book]
+}
+
+struct BookUpdateCommand: Command {
+  // Result of this command will be an updated book.
+  typealias Output = Book
+
+  // Let's pass the entire model to the command to simplify this example.
+  // Ideally we wouldn't do that because a command is supposed to be as simple
+  // as possible, only with attributes that are needed for handler.
+  let book: Book
+}
+```
+
+**Note** that any type can play a role of `Output`, so if we want to add a date
+to our `BookUpdateCommand` it could look like the following:
```

Comment:
From:
> **Note** that any type can play a role of `Output`, so if we want to add a date
to our `BookUpdateCommand` it could look like the following:

To:
> **Note** that any type can play the role of `Output`, so if we want to add a date
to our `BookUpdateCommand` it could look like the following: